crates/ark/src/r_task.rs

```diff
@@ -146,7 +146,9 @@ where
     T: 'env + Send,
 {
     // Escape hatch for unit tests
-    if stdext::IS_TESTING {
+    // In integration tests with dummy frontends, we have a "real" RMain and want to
+    // go through the standard r-task path
+    if stdext::IS_TESTING && !RMain::is_initialized() {
```

Comment:
Our integration tests that start up a dummy frontend happened to never run any `r_task()`s. If they did, we would have seen `R has already been initialized` as an error.

Basically, in integration tests with our dummy frontend we _actually have RMain_ and starting that RMain already called `Rf_initialize_R()`, but we try and do it again here in `r_test_init()`, resulting in that above error. We want to bypass this particular usage of `stdext::IS_TESTING` in integration tests and actually treat it like production.

It is confusing and I'm hoping to move us over to nextest soon which may help simplify this